samples/azure/performance-tuning-asb/sample.md

```diff
@@ -0,0 +1,133 @@
+
+Individual sends are not awaited, instead the entire batch is awaited. Code execution will continue when all batches have been sent.
+
+Snippet: fast-send
+
+### Slow Atomic Receiver
+
+In the slow atomic receiver sample, everything is configured wrong:
+ * Only 1 factory is defined, this means that only 1 tcp connection is maintained with the broker.
+ * Prefetching is turned off, meaning that the receiver will fetch message by message
+ * No concurrent operations allowed, messages are processed one by one.
+
+Snippet: slow-atomic-receiver-config
+
+### Fast Atomic Receiver
+
+The fast atomic receiver is configured much better
+ * Multiple connections are established with the broker.
+ * Each connection is matched with exactly 1 receive client.
+ * Ensures that at least 16 connections are established when using partitioned queues, as each partitioned queue consists of 16 partitions. Depending on available bandwidth configure more until the network saturates.
+ * TransactionMode AtomicSendsWithReceive will execute complete operations in a serializable transaction, which means one by one. Therefore it is not beneficial to allow high per receiver concurrency. A value of 2 allows one receive to start while another is allowed to complete. The more time execution takes the higher this value can be, but never use really high values here.
+ * PrefetchCount is set to a relatively low number, as per receiver concurrency and throughput is low it makes no sense to load many messages at once.
+
+Snippet: fast-atomic-receiver-config
+
+### Fast Atomic Sender Receiver
+
+The fast atomic sender receiver scenario is similar to the fast atomic receiver scenario, except:
+ * It established twice the amount of connections, so that the receive clients and send clients can leverage different connections.
+
+Snippet: fast-atomic-send-receive-config
+
+### Slow Non Atomic Receiver
+
+In the slow none atomic receiver sample, everything is configured wrong:
+ * Only 1 factory is defined, this means that only 1 tcp connection is maintained with the broker.
+ * Prefetching is turned off, meaning that the receiver will fetch message by message
+ * No concurrent operations allowed, messages are processed one by one.
+
+Snippet: slow-non-atomic-receiver-config
+
+### Fast Non Atomic Receiver
+
+The fast non atomic receiver is configured much better
+ * Multiple connections are established with the broker.
+ * Each connection is matched with exactly 1 receive client.
+ * Ensures that at least 16 connections are established when using partitioned queues, as each partitioned queue consists of 16 partitions. Depending on available bandwidth configure more until the network saturates.
+ * TransactionMode ReceiveOnly can batch message completions, allowing receive operations to work independently. Therefore it is very beneficial to allow high per receiver concurrency. For example 128 operations per receiver if the operation can handle this kind of concurrency.
```

Comment:
That really depends on the code inside the handler, a db connection may not like 128 concurrent operations, while an in memory operation on a concurrentdictionary may